Source/NSStoryboard.m

```diff
@@ -0,0 +1,975 @@
+                }
+
+              NSArray *customObjects = [document nodesForXPath: @"//objects/customObject" error: NULL];
+              NSEnumerator *coen = [customObjects objectEnumerator];
+              NSXMLElement *coel = nil;
+              while ((coel = [coen nextObject]) != nil)
+                {
+                  NSXMLNode *attr = [coel attributeForName: @"sceneMemberID"];
+                  if ([[attr stringValue] isEqualToString: @"firstResponder"])
+                    {
+                      NSXMLNode *customClassAttr = [coel attributeForName: @"customClass"];
+                      NSXMLNode *idAttr = [coel attributeForName: @"id"];
+                      NSString *originalId = [idAttr stringValue];
+
+                      [idAttr setStringValue: @"-1"]; // set to first responder id
+                      [customClassAttr setStringValue: @"FirstResponder"];
+
+                      // Actions
+                      NSArray *cons = [document nodesForXPath: @"//action" error: NULL];
+                      NSEnumerator *consen = [cons objectEnumerator];
+                      NSXMLElement *celem = nil;
+
+                      while ((celem = [consen nextObject]) != nil)
+                        {
+                          NSXMLNode *targetAttr = [celem attributeForName: @"target"];
+                          NSString *val = [targetAttr stringValue];
+                          if ([val isEqualToString: originalId])
+                            {
+                              [targetAttr setStringValue: @"-1"];
+                            }
+                        }
+
+                      // Outlets
+                      cons = [document nodesForXPath: @"//outlet" error: NULL];
+                      consen = [cons objectEnumerator];
+                      celem = nil;
+                      while ((celem = [consen nextObject]) != nil)
+                        {
+                          NSXMLNode *attr = [celem attributeForName: @"destination"];
+                          NSString *val = [attr stringValue];
+                          if ([val isEqualToString: originalId])
+                            {
+                              [attr setStringValue: @"-1"];
+                            }
+                        }
+                    }
+                }
+
+              // Delete extra firstResponder...
+              /*
+              NSArray *ar = [document nodesForXPath: @"//customObject[@sceneMemberID=\"firstResponder\"]"
+                                              error: NULL];
+              if ([ar count] > 0)
+                {
+                  id node = [ar objectAtIndex: 0];
+                  [node detach];
+                }
+              */
+              
+              // Create document...
+              [_scenesMap setObject: document
+                             forKey: sceneId];
+              
+              // Map controllerId's to scenes...
+              if (controllerId != nil)
```

Comment:
Does this mean there will ever by only one controller in any story board file? The above code suggests that this should rather be called for every controller we find in the file.

Reply:
There will actually only be one controller in each scene.   I tried multiple ways to prove this to myself when working in Xcode.  Additionally, if you look at the "compiled" storyboard which, basically ends up being a bunch of nibs, there is always one controller in each scene.   the purpose of this variable is to make each controllerId to it's corresponding scene so that we can instantiate the proper objects when instantiateWithControllerId: is called.

Reply:
If this is the case, the code above should express this better. Maybe have a Sparte method for each  scene transformation and another method to get the one controller for each scene?

Reply:
Done.